sql/catalyst/src/main/java/org/apache/spark/sql/execution/UnsafeExternalRowSorter.java

```diff
@@ -159,6 +154,12 @@ public boolean hasNext() {
         @Override
         public UnsafeRow next() {
           try {
+            if (!alreadyCalculated) {
```

Comment:
When `hasNext` is called, doesn't `sortedIterator` return no element anymore since we haven't added rows into it?

Reply:
Yes. There are mistakes here, I will change.